### PR TITLE
Different improvements

### PR DESCRIPTION
**Pull request by sponkrashin**

### CODE REVIEW

File: server/Arcadia.Assistant.Web/Arcadia.Assistant.Web.csproj [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<Project Sdk="Microsoft.NET.Sdk.Web">
+﻿<Project Sdk="Microsoft.NET.Sdk.Web">
```
Comment (santee) [inline server/Arcadia.Assistant.Web/Arcadia.Assistant.Web.csproj]: cool change.

File: server/Arcadia.Assistant.Web/Controllers/DownloadWebController.cs [hunk 2]
```diff
@@ -37,10 +39,29 @@ public class DownloadWebController : Controller
...
-                });
+            var userAgent = this.Request.Headers[HeaderNames.UserAgent];
+            var deviceType = this.GetDeviceTypeByUserAgent(userAgent);
+
+            return deviceType == DeviceType.Android
+                ? this.RedirectToAction("DownloadForAndroid")
+                : this.RedirectToAction("DownloadForIos");
+        }
+
+        [Route("download-for-android")]
```
[the remark this answers is not in the document]
Comment (sponkrashin) [reply server/Arcadia.Assistant.Web/Controllers/DownloadWebController.cs]: Done.

File: server/Arcadia.Assistant.Web/Controllers/DownloadWebController.cs [hunk 3]
```diff
@@ -37,10 +39,29 @@ public class DownloadWebController : Controller
...
+        [HttpGet]
+        public IActionResult DownloadForAndroid()
+        {
+            return this.View();
+        }
+
+        [Route("download-for-ios")]
+        [HttpGet]
+        public IActionResult DownloadForIos()
```
[the remark this answers is not in the document]
Comment (santee) [reply server/Arcadia.Assistant.Web/Controllers/DownloadWebController.cs]: GetFile is an action. Put it in a separate controller? What other common methods?

[the remark this answers is not in the document]
Comment (sponkrashin) [reply server/Arcadia.Assistant.Web/Controllers/DownloadWebController.cs]: Honestly, `GetFile` is just a method that returns `IActionResult` :)

[the remark this answers is not in the document]
Comment (sponkrashin) [reply server/Arcadia.Assistant.Web/Controllers/DownloadWebController.cs]: I'm not sure if you like the result, but I did it.

File: server/Arcadia.Assistant.Web/Controllers/DownloadIosWebController.cs [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+{
+    using System.Text;
+
+    using Microsoft.AspNetCore.Mvc;
+
+    using Arcadia.Assistant.Web.Configuration;
+    using Arcadia.Assistant.Web.Models;
+    using Arcadia.Assistant.Web.Properties;
+
+    public class DownloadIosWebController : Controller
```
Comment (santee) [inline server/Arcadia.Assistant.Web/Controllers/DownloadIosWebController.cs]: IgnoreApi?

**Comment by sponkrashin** [comment]
Yes and no. It looks neater, but we have to remember about controller names, actions names from them and control correctness of links in case of names refactoring.

File: server/Arcadia.Assistant.Web/Controllers/DownloadIosWebController.cs (hunk 4 above)
Comment (sponkrashin) [reply server/Arcadia.Assistant.Web/Controllers/DownloadIosWebController.cs]: Done.

**Pull request merged**